fullrt/dht.go

```diff
@@ -777,10 +777,14 @@ func (dht *FullRT) Provide(ctx context.Context, key cid.Cid, brdcst bool) (err e
 		return fmt.Errorf("invalid cid: undefined")
 	}
 	keyMH := key.Hash()
-	logger.Debugw("providing", "cid", key, "mh", internal.LoggableProviderRecordBytes(keyMH))
+	mhHash := internal.Sha256Multihash(keyMH)
```

Comment:
do you think it might be a good idea to add a salt for this hash? this way there's another layer of protection that could be used from the provider? 
but i can see the other side where if we want to make this easily reproducible, it would be better to leave out the additional step (-:

Reply:
If required, we could add a constant salt (e.g to specify the purpose of the hash).

However, the _secret_ to access the content is already the CID, adding a random salt means that knowledge of both the CID and the salt is required to find the content. If we add a random salt here, it would have to be shared along with the CID, thus it wouldn't add much security. (in other words, it would only make the _secret_ longer)

Reply:
We should consider adding a constant salt (or prefix) to the spec and here - so that the derived hashes are in a known specific domain that doesn't easily have collisions with other things.